### PR TITLE
Deploy/3/rds

### LINKED ISSUE #3: Assignment 2
Create a VPC with one private and public subnet.
Create an RDS MySQL in the public subnet of the VPC. Stop the database when not in use.
While creating RDS, keep the engine type as MySQL, edition as MySQL Community, instance type as db.t3.micro and storage type as gp2
Connect to the RDS MySQL and create a table "logs".
Move the RDS MySQL to a private subnet.
Create a flask app with a single route which connects to the DB. The app should insert a record with a timestamp into the DB for each API call.
Deploy the flask app in an autoscaling group (Set the desired size as 2, and enable Spot Instance when setting up Auto Scaling).
Provision an AWS Application load balancer which connects to the Autoscaling group.
Create a Lambda function which scales down the autoscaling group to zero on every Saturday and scales up to 1 on every Monday morning.

### PR DESCRIPTION
**Pull request by saichandupaleti**
Created a VPC with one public and one private subnet.
Created an RDS MySQL instance (db.t3.micro, gp2) in the public subnet.
Created a logs table in the RDS instance.
Moved the RDS instance to the private subnet
Created an EC2 instances in the same VPC under public subnet to access RDS
Developed and deployed a Flask app that inserts a timestamp into the logs table on each API call.
Setup an auto scaling group of desired size 2 and an application load balancer for this scaling group.
Created lambda functions to scale down on Saturdays and scale up to one on mondays

Inside EC2-Instance:
image
Screenshot 2024-05-22 214617

Using autoscaling groups and application load balancers:
Screenshot 2024-05-22 182133

### CODE REVIEW

File: README.md
Comment (nilesh-tiger) [inline README.md:1]: Add appropriate screenshots and implementation steps to readme. Just like you have it in the PR description

File: flaskapp/flask_db.py
Comment (nilesh-tiger) [inline flaskapp/flask_db.py:1]: Add docstrings and proper documentation

File: flaskapp/test_boto.py
Comment (nilesh-tiger) [inline flaskapp/test_boto.py:1]: Add documentation

Comment (nilesh-tiger) [inline flaskapp/test_boto.py:1]: This should be a part of src

File: tasks.py
Comment (nilesh-tiger) [inline tasks.py:1]: add to appropriate folder, should be part of src

**Review by nilesh-tiger — COMMENTED** [review]
Implement the lambda function to have auto- scale the instance as given in the assignment

File: README.md
Comment (saichandupaleti) [reply README.md:1]: Ok Nilesh

File: flaskapp/flask_db.py
Comment (saichandupaleti) [reply flaskapp/flask_db.py:1]: Added Nilesh

File: flaskapp/test_boto.py
Comment (saichandupaleti) [reply flaskapp/test_boto.py:1]: this code is for assignment 1 nilesh i will remove it, sorry for the issue

Comment (saichandupaleti) [reply flaskapp/test_boto.py:1]: same above file nilesh,i will remove it

File: tasks.py
Comment (saichandupaleti) [reply tasks.py:1]: done nilesh

**Review by nilesh-tiger — APPROVED** [review]
good work overall

**Pull request merged**